darts/models/forecasting/tide_model.py

```diff
@@ -0,0 +1,526 @@
+        if self.future_cov_dim > 0:
+            x_dynamic_covariates = torch.cat(
+                [x_future_covariates, x[:, :, -self.future_cov_dim :]], dim=1
+            )
+
+            # project input features across all input time steps
+            x_dynamic_covariates_proj = self.feature_projection(x_dynamic_covariates)
+
+        else:
+            x_dynamic_covariates = None
+            x_dynamic_covariates_proj = None
+
+        # extract past covariates, if they exist
+        if self.input_dim - self.output_dim - self.future_cov_dim > 0:
+            x_past_covariates = x[:, :, self.output_dim : -self.future_cov_dim]
+        else:
+            x_past_covariates = None
+
+        # setup input to encoder
+        encoded = [
+            x_lookback,
+            x_past_covariates,
+            x_dynamic_covariates_proj,
+            x_static_covariates,
+        ]
+        encoded = [t.flatten(start_dim=1) for t in encoded if t is not None]
+        encoded = torch.cat(encoded, dim=1)
+
+        # encoder, decode, reshape
+        encoded = self.encoders(encoded)
+        decoded = self.decoders(encoded)
+        # decoded = decoded.view(-1, self.output_chunk_length, self.decoder_output_dim)
+
+        # get view that is batch size x output chunk length x self.decoder_output_dim x nr params
+        decoded = decoded.view(
+            -1, self.output_chunk_length, self.decoder_output_dim, self.nr_params
+        )
+
+        # stack and temporally decode with future covariate last output steps
+        temporal_decoder_input = [
+            decoded.flatten(start_dim=2),
+            x_dynamic_covariates_proj[:, -self.output_chunk_length :, :]
+            if self.future_cov_dim > 0
+            else None,
+        ]
+        temporal_decoder_input = [t for t in temporal_decoder_input if t is not None]
+
+        temporal_decoder_input = torch.cat(temporal_decoder_input, dim=2)
+        temporal_decoded = self.temporal_decoder(temporal_decoder_input)
+
+        # pass x_lookback through self.lookback_skip but swap the last two dimensions
+        # this is needed because the skip connection is applied across the input time steps
+        # and not across the output time steps
+        skip = self.lookback_skip(x_lookback.transpose(1, 2)).transpose(1, 2)
+
+        # add skip connection
+        y = temporal_decoded + skip.reshape_as(
+            temporal_decoded
+        )  # skip.view(temporal_decoded.shape)
+
+        y = y.view(-1, self.output_chunk_length, self.output_dim, self.nr_params)
+
+        return y
+
+
+class TiDEModel(MixedCovariatesTorchModel):
+    def __init__(
+        self,
+        input_chunk_length: int,
+        output_chunk_length: int,
+        num_encoder_layers: int = 1,
+        num_decoder_layers: int = 1,
+        decoder_output_dim: int = 16,
+        hidden_size: int = 512,
+        temporal_width: int = 128,
+        temporal_decoder_width: int = 64,
```

Comment:
it would be nice to use similar parameter names as in the hyperparameter appendix from the paper.